### ISSUE TITLE
IImage.Save (SkiaImage) not implemented

### ISSUE THREAD

**Issue by StuffOfInterest** [dotnet/Microsoft.Maui.Graphics#437]
This is a follow-on from #436.  Building some code to replace `System.Drawing.Common` image manipulation for a ASP.NET 6 website running on Windows.  Tried to take an `IImage` over to a byte array via a memory stream as shown [here](https://docs.microsoft.com/en-us/dotnet/maui/user-interface/graphics/images).  The exact code doing this work is:

```cs
public static byte[] ConvertImageToByteArray(IImage imageToConvert, ImageFormat formatOfImage)
{
    byte[] ret;

    using (var ms = new MemoryStream())
    {
        imageToConvert.Save(ms, formatOfImage);
        ret = ms.ToArray();
    }

    return ret;
}
```

The `IImage` was extracted from a byte array using `SkiaImage` as shown in #436.  The call to `.Save()` ends up with a `PlatformNotImplementedException`.

<img width="638" alt="image" src="https://user-images.githubusercontent.com/8578647/170865264-649c1ecb-3537-4d8e-b652-35945d9bacd8.png">

Being that I'm using version 6.0.300 of both the `Microsoft.Maui.Graphics` and `Microsoft.Maui.Graphics.Skia` NuGet packages, I thought MAUI was up to feature complete status.  Did this method get overlooked is is Skia not intended to support Windows?

**Comment by sharpSteff** [comment]
IMO it's not ok to tell devs to migrate from System.Drawing.Common to Microsoft.Maui.Graphics when this lib is clearly not ready for production

**Comment by RChrisCoble** [comment]
We're just using SkiaSharp directly, as that's the primary rendering engine behind MAUI.Graphics anyway.  Our primary UC has us using SkiaSharp in Blazor WASM (with fantastic success), then we'll move onto other platforms.  Skia/SkiaSharp supports as many platforms as you would need.  If MAUI.Graphics gets its act together and additionally supports Blazor Server (which is required for Blazor Hybrid support) that would give us a compelling reason to move off of SkiaSharp.

**Comment by sharpSteff** [comment]
Then what's the purpose of Microsoft.Maui.Graphics, when it's using SkipSharp any way?

**Comment by RChrisCoble** [comment]
It's an abstraction layer, allowing other rendering implementations to be used when appropriate in the future.  It may not make sense for MAUI to require delivery of Google's Skia and MS's SkiaSharp on every platform MAUI wants to support.

**Comment by sharpSteff** [comment]
> It's an abstraction layer, allowing other rendering implementations to be used when appropriate in the future.  It may not make sense for MAUI to require delivery of Google's Skia and MS's SkiaSharp on every platform MAUI wants to support.

Yes exactly, e.g on windows. GDI+ is fast and solid. It's a mess that there is no GDI+ Implementation yet. (Source code is available but not published as a Nuget)

**Comment by swharden** [comment]
Are there plans to publish the GDI package?

I'd be more inclined to use and contribute to it if it were published on NuGet with the other packages

**Comment by sharpSteff** [comment]
@swharden I've created an issue for this #489

**Comment by lindexi** [comment]
@StuffOfInterest https://github.com/dotnet/Microsoft.Maui.Graphics/pull/475
